components/hab/install.sh

```diff
@@ -1,58 +1,318 @@
 #!/bin/sh
+#
+# Copyright (c) 2010-2016 Chef Software, Inc. and/or applicable contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
 # Fails on unset variables & whenever a command returns a non-zero exit code.
-set -eu
+set -eux
 # If the variable `$DEBUG` is set, then print the shell commands as we execute.
 if [ -n "${DEBUG:-}" ]; then set -x; fi
 
-# Download URL for a `core/hab` Habitat slim archive
-_url='https://api.bintray.com/content/habitat/stable/linux/x86_64/hab-$latest-x86_64-linux.tar.gz'
-_q="?bt_package=hab-x86_64-linux"
-hab_url="${_url}$_q"
-# Download URL for the shasum digest
-sha_url="${_url}.sha256sum$_q"
+# Borrowed from Omnitruck.
+# helpers.sh
+############
+# This section has some helper functions to make life easier.
+#
+# Outputs:
+# $tmp_dir: secure-ish temp directory that can be used during installation.
+############
+
+# Check whether a command exists - returns 0 if it does, 1 if it does not
+exists() {
+  if command -v $1 >/dev/null 2>&1
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# Output the instructions to report bug about this script
+report_bug() {
+  echo ""
+  echo "Please file a Bug Report at https://github.com/habitat-sh/habitat/issues/new"
+  echo ""
+  echo "Please include as many details about the problem as possible i.e., how to reproduce"
+  echo "the problem (if possible), type of the Operating System and its version, etc.,"
+  echo "and any other relevant details that might help us with troubleshooting."
+  echo ""
+}
+
+checksum_mismatch() {
+  echo "Package checksum mismatch!"
+  report_bug
+  exit 1
+}
+
+unable_to_retrieve_package() {
+  echo "Unable to retrieve a valid package!"
```

Comment:
We should include the reason we failed here and not just the outcome of the install. It looks like we've failed because we were unable to locate a suitable means to retrieve a remote asset over HTTP. We should also include some instructions like, "Hey dummy! Install wget or something!", but... maybe more friendly ;)